pkg/block/factory/build.go

```diff
@@ -148,24 +149,27 @@ func buildGSAdapter(ctx context.Context, params params.GS) (*gs.Adapter, error)
 }
 
 func buildAzureAdapter(params params.Azure) (*azure.Adapter, error) {
-	p, credentials, err := BuildAzureServiceClient(params)
+	p, err := BuildAzureServiceClient(params)
 	if err != nil {
 		return nil, err
 	}
-	return azure.NewAdapter(*p, credentials), nil
+	return azure.NewAdapter(p), nil
 }
 
-func BuildAzureServiceClient(params params.Azure) (*service.Client, *aztables.SharedKeyCredential, error) {
-	cred, err := azblob.NewSharedKeyCredential(params.StorageAccount, params.StorageAccessKey)
-	if err != nil {
-		return nil, nil, fmt.Errorf("invalid credentials: %w", err)
+func BuildAzureServiceClient(params params.Azure) (*service.Client, error) {
+	url := fmt.Sprintf(azure.AzURLTemplate, params.StorageAccount)
+	options := service.ClientOptions{ClientOptions: azcore.ClientOptions{Retry: policy.RetryOptions{TryTimeout: params.TryTimeout}}}
+	if params.StorageAccessKey != "" {
```

Comment:
We should change the documentation to reflect this

Reply:
Opened https://github.com/treeverse/lakeFS/issues/5044